knocker.py

```diff
@@ -26,8 +29,13 @@ def main():
     parser.add_argument('ports', nargs='+', type=int, help='Ports to knock')
     parser.add_argument('-d', '--delay', type=int, default=200,
                         help='Milliseconds between each knock')
+    parser.add_argument('-p', '--protocol', default='tcp')
 
     args = parser.parse_args()
 
+    if not(args.protocol.lower() == 'tcp' or args.protocol.lower() == 'udp'):
```

Comment:
A better way to check the options would be this:`if args.protocol.lower() not in ['tcp', 'udb']`. But, argparse has a choices options: https://docs.python.org/3/library/argparse.html#choices.